### ISSUE TITLE
Rusty-tags should optionally accept a file as a starting point to search for Cargo.toml

### ISSUE THREAD

**Issue by TyOverby** [dan-t/rusty-tags#20]
I do most of my editing from a directory that contains multiple rust projects.  This means that when vim runs `rusty-tags vi` it starts in a directory that isn't even a rust project!

the rusty-tags binary should accept an optional file path as a starting point for the Cargo.toml search (and subsequent tag generation).

**Comment by dan-t** [comment]
I don't quite understand your use case.

So you're editing a Rust file that isn't part of a cargo project, right? But how should `rusty-tags` create a correct tags file without a `Cargo.toml`?

Or does the Rust file belong to a cargo project? But how is the cargo project organized, how does this Rust file belong to it?

**Comment by TyOverby** [comment]
I have a project that looks like this:

```
* My Big Project
    * Crate 1
        * src
            * lib.rs
        * Cargo.toml
    * Crate 2
        * src
            * lib.rs
        * Cargo.toml
```

I always open up vim from inside My Big Project which *isn't* a crate.  However, I'm editing files that *are* inside of a crate.

**Comment by dan-t** [comment]
Where you open vim shouldn't matter. Are use using the vim configuration described in the README?
If you're e.g. editing `lib.rs` of crate 1 and writing it, than `rusty-tags` should find the `Cargo.toml` of crate 1 and put a `rusty-tags.vi` beside of it.

Oh wait, do you have `set autochdir` in your `.vimrc`. If not, then the current working directory
isn't changed when editing a file, and `rusty-tags` might not find the `Cargo.toml`.

Is this your problem? Or do you want to call `rusty-tags` from the big project directory and
creating tags for crate 1 and 2?

**Comment by TyOverby** [comment]
I don't use autochdir because I like shelling out from the root directory, so that's my problem.

**Comment by dan-t** [comment]
I just released `rusty-tags 0.6.8` which has a `--start-dir` option.

You can now use these vim settings and it should work without `autochdir`:
    setlocal tags=./rusty-tags.vi;/
    autocmd BufWrite *.rs :silent exec "!rusty-tags vi --start-dir=" . expand('%:p:h') . "&"

**Comment by TyOverby** [comment]
Thanks!
